src/java.base/share/classes/java/io/ObjectStreamClass.java

```diff
@@ -1633,13 +1667,13 @@ private static ObjectStreamField[] getDeclaredSerialFields(Class<?> cl)
     {
         ObjectStreamField[] serialPersistentFields = null;
         try {
-            Field f = cl.getDeclaredField("serialPersistentFields");
+            Field f = getDeclaredField(cl, ObjectStreamField[].class, "serialPersistentFields");
```

Comment:
True.

On the other hand, what about a .class file that includes all of (pseudo-Java)

```
    private static final Object              serialPersistentFields = new ObjectStreamField[0];
    private static final Cloneable           serialPersistentFields = new ObjectStreamField[0];
    private static final Serializable        serialPersistentFields = new ObjectStreamField[0];
    private static final ObjectStreamField[] serialPersistentFields = new SubclassOfObjectStreamField[0];
```

Which one is the "preferred" field?

Perhaps the Java Object Serialization Specification should simply prohibit multiple `serialPersistentFields`.

Reply:
> This can technically be a breaking change, as it was supported for the field to have a declared type that is assignable from `ObjectStreamField[]`, as long as it held an `ObjectStreamField[]` instance at runtime, even if it wasn’t officially supported.
> 
> ```java
> class Example implements Serializable {
> 	// This used to work before this patch in OpenJDK
> 	private static final Object serialPersistentFields = new ObjectStreamField[] {
> 		// ...
> 	};
> 
> 	// ...
> }
> ```

That sort of behavior change would require a CSR; marking the PR accordingly.